ziggurat-core-geoip/src/geoutils.rs

```diff
@@ -0,0 +1,49 @@
+use crate::geoip::Location;
+
+/// Calculate the distance between two points using Vincenty's inverse formula.
+/// The distance is returned in meters.
+pub fn geo_get_distance_between(location1: Location, location2: Location) -> f64 {
+    let loc1 = geoutils::Location::new(location1.latitude, location1.longitude);
+    let loc2 = geoutils::Location::new(location2.latitude, location2.longitude);
+
+    // Calculate the distance between the two points using Vincenty's inverse formula.
+    loc1.distance_to(&loc2).map(|d| d.meters()).unwrap_or(0.0)
+}
+
+/// Check if the location is in the circle of radius meters.
+pub fn geo_is_in_circle(location1: Location, location2: Location, radius: f64) -> bool {
+    let distance = geo_get_distance_between(location1, location2);
+    distance <= radius
```

Comment:
https://docs.rs/geoutils/latest/geoutils/struct.Location.html#method.is_in_circle
we can use this directly

but we created a wrapper around `geoutils::Location`, didn't we? we could use that directly instead of creating our own `Location` struct, but cons is the extra depedency

Reply:
Exactly, that was the reason to wrap - I think it's better to wrap structure and leave all geo-crates dependencies to be managed by ziggurat-core instead of letting each package manage on its own.